eis_toolkit/vector_processing/proximity_computation.py

```diff
@@ -0,0 +1,76 @@
+from numbers import Number
+
+import geopandas as gpd
+import numpy as np
+from beartype import beartype
+from beartype.typing import Literal, Optional, Tuple, Union
+from rasterio import profiles
+
+from eis_toolkit.transformations.linear import _min_max_scaling
+from eis_toolkit.vector_processing.distance_computation import distance_computation
+
+
+@beartype
+def proximity_computation(
+    geodataframe: gpd.GeoDataFrame,
+    raster_profile: Union[profiles.Profile, dict],
+    maximum_distance: Number,
+    scaling_method: Literal["linear"],
+    scale_range: Optional[Tuple[Number, Number]] = None,
+) -> np.ndarray:
+    """Compute proximity to the specified polygons based on the chosen scaling method.
+
+    Args:
+         geodataframe: The GeoDataFrame with geometries to determine distance to.
+         raster_profile: The raster profile of the raster in which the distances
+                         to the nearest geometry are determined.
+         max_distance: The maximum distance in the output array.
+         scaling_method: method of scaling
+                         linear: linear,
+         scaling_range: a tuple of maximum value and minimum value to scale the distances.
```

Comment:
Here are my suggestions how to modify this docstring.
```suggestion
    """Compute proximity to the nearest geometries.

    Args:
         geodataframe: The GeoDataFrame with geometries to determine proximity to.
         raster_profile: The raster profile of the raster in which the distances to the nearest geometry are determined.
         max_distance: The distance in the output array beyond which proximity is considered 0.
         scaling_method: Scaling method used to produce the proximity values. Defaults to 'linear'.
         scaling_range: Min and max values used for scaling the proximity values. Defaults to (1, 0).
```